clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -273,6 +276,7 @@ public void authenticationFailed(String id, long now, AuthenticationException ex
         nodeState.state = ConnectionState.AUTHENTICATION_FAILED;
         nodeState.lastConnectAttemptMs = now;
         updateReconnectBackoff(nodeState);
+        connectingNodes.remove(id);
```

Comment:
Can we get here with id in `connectingNodes`?

Reply:
looks like the line is no longer needed to pass all the tests !